x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/EnrichMultiNodeIT.java

```diff
@@ -0,0 +1,177 @@
+            assertThat(searchResponse.getHits().getTotalHits().value, equalTo(8L));
+        }
+
+        ListEnrichPolicyAction.Response response =
+            client().execute(ListEnrichPolicyAction.INSTANCE, new ListEnrichPolicyAction.Request()).actionGet();
+        assertThat(response.getPolicies().size(), equalTo(numPolicies));
+
+        for (int i = 0; i < numPolicies; i++) {
+            String policyName = POLICY_NAME + i;
+            client().execute(DeleteEnrichPolicyAction.INSTANCE, new DeleteEnrichPolicyAction.Request(policyName)).actionGet();
+        }
+
+        response = client().execute(ListEnrichPolicyAction.INSTANCE, new ListEnrichPolicyAction.Request()).actionGet();
+        assertThat(response.getPolicies().size(), equalTo(0));
+    }
+
+    public void testEnrich() {
+        List<String> nodes = internalCluster().startNodes(3);
+        List<String> keys = createSourceIndex(64);
+        createAndExecutePolicy();
+        createPipeline();
+        enrich(keys, randomFrom(nodes));
+    }
+
+    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/pull/43311")
+    public void testEnrichDedicatedIngestNode() {
+        internalCluster().startNode();
+        Settings settings = Settings.builder()
+            .put(Node.NODE_MASTER_SETTING.getKey(), false)
+            .put(Node.NODE_DATA_SETTING.getKey(), false)
+            .put(Node.NODE_INGEST_SETTING.getKey(), true)
+            .build();
+        String ingestOnlyNode = internalCluster().startNode(settings);
+
+        List<String> keys = createSourceIndex(64);
+        createAndExecutePolicy();
+        createPipeline();
+        enrich(keys, ingestOnlyNode);
+    }
+
+    private static void enrich(List<String> keys, String coordinatingNode) {
+        int numDocs = 256;
+        BulkRequest bulkRequest = new BulkRequest("my-index");
+        for (int i = 0; i < numDocs; i++) {
+            IndexRequest indexRequest = new IndexRequest();
+            indexRequest.id(Integer.toString(i));
+            indexRequest.setPipeline(PIPELINE_NAME);
+            indexRequest.source(Map.of(KEY_FIELD, randomFrom(keys)));
+            bulkRequest.add(indexRequest);
+        }
+        BulkResponse bulkResponse = client(coordinatingNode).bulk(bulkRequest).actionGet();
+        assertThat("Expected no failure, but " + bulkResponse.buildFailureMessage(), bulkResponse.hasFailures(), is(false));
+
+        for (int i = 0; i < numDocs; i++) {
+            GetResponse getResponse = client().get(new GetRequest("my-index", Integer.toString(i))).actionGet();
+            Map<String, Object> source = getResponse.getSourceAsMap();
+            assertThat(source.size(), equalTo(1 + DECORATE_FIELDS.length));
+            for (String field : DECORATE_FIELDS) {
+                assertThat(source.get(field), notNullValue());
+            }
+        }
+        client().admin().indices().delete(new DeleteIndexRequest("my-index")).actionGet();
```

Comment:
should this be done in an annotated `@before` method? What if assert fails above w/ null?

Reply:
Actually this is not needed; the base test class already takes care of this.